sdcm/nemesis.py

```diff
@@ -462,6 +462,35 @@ def disrupt_nodetool_decommission(self, add_node=True):
                 if new_node:
                     new_node.running_nemesis = None
 
+    def disrupt_nodetool_seed_decommission(self, add_node=True):
+        def update_seed_provider(_seed_address):
+            seed_provider = [dict(class_name='org.apache.cassandra.locator.SimpleSeedProvider',
+                                  parameters=[dict(seeds=seed_address)])]
+            for node in self.cluster.nodes:
+                node.patch_scylla_yaml(seed_provider=seed_provider)
+
+        if len(self.cluster.seed_nodes) < 2:
+            raise UnsupportedNemesis("To running seed decommission the cluster must contains at least 2 seed nodes")
+        if not self.target_node.is_seed:
+            self.target_node = random.choice(self.cluster.seed_nodes)
+        target_seed_idx = self.cluster.seed_nodes.index(self.target_node)
+        self.log.info(f"Removing the seed node in index '{target_seed_idx}' from seed nodes list")
+        self.cluster.seed_nodes.pop(target_seed_idx)
+        self.target_node.is_seed = False
+
+        seed_address = ",".join([node.ip_address for node in self.cluster.seed_nodes])
+        update_seed_provider(_seed_address=seed_address)
+
+        nodes = self.cluster.nodes
+        self.disrupt_nodetool_decommission(add_node=add_node)
+        new_seed_node = set(self.cluster.nodes) - set(nodes)
```

Comment:
won't it be easier to return the new node from `disrupt_nodetool_decommission`, and then you'll have less logic ?

Reply:
Done